testsuite/integration/expansion/src/test/java/org/wildfly/test/integration/observability/micrometer/ConflictingPrometheusContextTestCase.java

```diff
@@ -0,0 +1,130 @@
+import org.junit.runner.RunWith;
+import org.wildfly.test.integration.observability.JaxRsActivator;
+
+@RunWith(Arquillian.class)
+@ServerSetup({MicrometerSetupTask.class})
+@DockerRequired
+@RunAsClient
+public class ConflictingPrometheusContextTestCase {
+    private static final ModelNode metricsExtension = Operations.createAddress("extension", "org.wildfly.extension.metrics");
+    private static final ModelNode metricsSubsystem = Operations.createAddress("subsystem", "metrics");
+
+    public static final ModelNode PROMETHEUS_REGISTRY_ADDRESS = Operations.createAddress(SUBSYSTEM, "micrometer", "registry", "prometheus");
+
+    private boolean metricsExtAdded = false;
+    private boolean metricsSubsystemAdded = false;
+
+    @ContainerResource
+    protected ManagementClient managementClient;
+
+    @Deployment
+    public static Archive<?> deploy() {
+        return ShrinkWrap.create(WebArchive.class, "micrometer-prometheus.war")
+                .addClasses(JaxRsActivator.class, MicrometerResource.class)
+                .addAsWebInfResource(CdiUtils.createBeansXml(), "beans.xml");
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        if (AssumeTestGroupUtil.isBootableJar() || AssumeTestGroupUtil.isWildFlyPreview() || isGalleon()) {
+            throw new AssumptionViolatedException("Not supported in this configuration");
+        }
+    }
+
+    private static boolean isGalleon() {
+        return System.getProperty("ts.layers") != null || System.getProperty("ts.galleon") != null;
+    }
+
+    @Test
+    @InSequence(1)
+    public void setupMetrics() throws IOException {
+        if (!Operations.isSuccessfulOutcome(executeRead(managementClient, metricsExtension))) {
+            executeOp(managementClient, Operations.createAddOperation(metricsExtension));
+            metricsExtAdded = true;
+        }
+
+        if (!Operations.isSuccessfulOutcome(executeRead(managementClient, metricsSubsystem))) {
+            executeOp(managementClient, Operations.createAddOperation(metricsSubsystem));
+            metricsSubsystemAdded = true;
+        }
+
+        ServerReload.executeReloadAndWaitForCompletion(managementClient);
+    }
+
+    @Test
+    @InSequence(2)
+    public void configureConflictingContexts() throws Exception {
+        ModelNode addOperation = Operations.createAddOperation(PROMETHEUS_REGISTRY_ADDRESS);
+        addOperation.get("context").set("${no.such.property:/metrics}");
+        addOperation.get("security-enabled").set("false");
+
+        try {
+            ModelNode response = managementClient.getControllerClient().execute(Operation.Factory.create(addOperation));
+            System.err.println(response);
+            assertTrue(response.asString(), response.get(ModelDescriptionConstants.FAILURE_DESCRIPTION)
+                .asString().contains("WFLYCTL0436"));
+//            Assert.fail("Adding a context that conflicts with WildFly Metrics should have failed.");
+        } catch (Exception e) {
```

Comment:
The catch block should be removed. This code should no longer throw an exception, so if it does it should propagate and fail the test.

Reply:
Good call. Done.